### PR TITLE
Add GitLab API

### PR DESCRIPTION
**Pull request by dreifachstein**
This is part of the effort to enable gitlab as the oauth provider for gerrit.

### CODE REVIEW

File: scribejava-apis/src/main/java/com/github/scribejava/apis/GitLabApi.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    public static GitLabApi instance() {
+        return InstanceHolder.INSTANCE;
+    }
+
+    @Override
+    public TokenExtractor<OAuth2AccessToken> getAccessTokenExtractor() {
+        return OAuth2AccessTokenJsonExtractor.instance();
+    }
```
Comment (kullfar) [inline scribejava-apis/src/main/java/com/github/scribejava/apis/GitLabApi.java]: it's not needed anymore. It's default extractor now, just rebase to the master, pls

File: scribejava-apis/src/test/java/com/github/scribejava/apis/examples/GitLabApiExample.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
+package com.github.scribejava.apis.examples;
+
+import java.util.Scanner;
+
+import com.github.scribejava.apis.GitLabApi;
+import com.github.scribejava.core.builder.ServiceBuilder;
+import com.github.scribejava.core.model.*;
+import com.github.scribejava.core.oauth.OAuth20Service;
+
+public class GitLabApiExample {
```
Comment (kullfar) [inline scribejava-apis/src/test/java/com/github/scribejava/apis/examples/GitLabApiExample.java]: "GitLabExample"
as others

**Comment by kullfar** [comment]
Looks clean and good! Thanks.
To merge this I'll need "user and test application credentials to run this example.
I do not have any opportunity to register myself in ALL API providers and to register a test application."
GitLabApiExample

Thanks in advance!

**Comment by kullfar** [comment]
I mean this
````
Next, you will need to provide me user and test application credentials to run this example.
I do not have any opportunity to register myself in ALL API providers and to register a test application.
(send them to that email https://github.com/scribejava/scribejava/blob/master/pom.xml#L48)
````
https://github.com/scribejava/scribejava/wiki/FAQ#when-will-you-support-insert_api_here-

File: scribejava-apis/src/main/java/com/github/scribejava/apis/GitLabApi.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+    @Override
+    public String getAccessTokenEndpoint() {
+        return baseUrl + ACCESS_TOKEN_ENDPOINT_PATH;
+    }
+
+    @Override
+    public Verb getAccessTokenVerb() {
+        return Verb.POST;
+    }
```
Comment (kullfar) [inline scribejava-apis/src/main/java/com/github/scribejava/apis/GitLabApi.java]: it's default now as well ;-)

**Comment by dreifachstein** [comment]
Updated and tested with clientId="b6e0e646d3b7a2d9d31d40c6aa31e1ee0d6f57f337f08eb04107108a68c6bcdd" and clientSecret="f6034f566c183293576e5b5451ace08b3e17fd14f2caab1febfbb8373c126a07".

File: scribejava-apis/src/test/java/com/github/scribejava/apis/examples/GitLabExample.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+import com.github.scribejava.apis.GitLabApi;
+import com.github.scribejava.core.builder.ServiceBuilder;
+import com.github.scribejava.core.model.OAuth2AccessToken;
+import com.github.scribejava.core.model.OAuthConstants;
+import com.github.scribejava.core.model.OAuthRequest;
+import com.github.scribejava.core.model.Response;
+import com.github.scribejava.core.model.Verb;
+import com.github.scribejava.core.oauth.OAuth20Service;
+
+public class GitLabExample {
```
Comment (kullfar) [inline scribejava-apis/src/test/java/com/github/scribejava/apis/examples/GitLabExample.java]: should be abstract. It's not intended to be instantiated

**Comment by kullfar** [comment]
Thanks for clientId and clientSecret, besides them I need user login/password to run the example.

**Comment by kullfar** [comment]
outdated. no login/password was got in two months ;-(

**Pull request closed without merging**